lib/active_record_shards/model.rb

```diff
@@ -9,13 +9,14 @@ def not_sharded
     end
 
     def is_sharded? # rubocop:disable Style/PredicateName
+      sharded_ivar = defined?(@sharded) ? @sharded : nil
```

Comment:
We already declare two ivars in a `base.after_initialize` callback. Can't we declare `@sharded` as `nil` there too?